bootstrap/kubeadm/api/v1alpha4/kubeadmconfigtemplate_webhook_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha4"
+)
+
+// These tests are written in BDD-style using Ginkgo framework. Refer to
+// http://onsi.github.io/ginkgo to learn more.
+
+func Test_KubeadmConfigTemplate_validates_creation_and_deletion_and(t *testing.T) {
+	cases := map[string]struct {
+		in        *bootstrapv1.KubeadmConfigTemplate
+		expectErr bool
+	}{
+		"accepts valid configuration": {
+			in: &bootstrapv1.KubeadmConfigTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "baz",
+					Namespace: "default",
+				},
+				Spec: bootstrapv1.KubeadmConfigTemplateSpec{
+					Template: bootstrapv1.KubeadmConfigTemplateResource{
+						Spec: bootstrapv1.KubeadmConfigSpec{},
+					},
+				},
+			},
+		},
+		"rejects bad Ignition configuration": {
```

Comment:
What about implementing checks for all the other bad ignition configurations:
- experimentalRetryJoin set
- inactive users
- gpt partitions

Reply:
This is already covered by tests in `bootstrap/kubeadm/api/v1alpha4/kubeadmbootstrapconfig_types_test.go`. This test ensures that validation is _called_, not how validation is _done_. I think it's better to test on lower levels rather than on higher levels. What do you think?

Reply:
ok, might be add a note then